openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/searchIndex/ElasticSearchIndexSink.java

```diff
@@ -113,101 +113,240 @@ public void write(List<?> entities, Map<String, Object> contextData) throws Sear
     }
   }
 
-  private void sendBulkRequest(List<DocWriteRequest<?>> requests, List<EntityError> entityErrorList)
+  private void sendBulkRequestAsync(
+      List<DocWriteRequest<?>> requests, List<EntityError> entityErrorList)
       throws SearchIndexException {
     BulkRequest bulkRequest = new BulkRequest();
     bulkRequest.add(requests);
 
-    int attempt = 0;
-    long backoffMillis = initialBackoffMillis;
+    try {
+      semaphore.acquire();
+      LOG.debug("Semaphore acquired. Available permits: {}", semaphore.availablePermits());
+      ((RestHighLevelClient) client.getClient())
+          .bulkAsync(
+              bulkRequest,
+              RequestOptions.DEFAULT,
+              new ActionListener<BulkResponse>() {
+                @Override
+                public void onResponse(BulkResponse response) {
+                  try {
+                    for (int i = 0; i < response.getItems().length; i++) {
+                      BulkItemResponse itemResponse = response.getItems()[i];
+                      if (itemResponse.isFailed()) {
+                        String failureMessage = itemResponse.getFailureMessage();
+                        String entityData = requests.get(i).toString();
+                        entityErrorList.add(
+                            new EntityError().withMessage(failureMessage).withEntity(entityData));
+                        LOG.warn("Bulk item failed: {}", failureMessage);
+                      }
+                    }
```

Comment:
What do you think of using a Stream here like this (caveat i have not tested it)

```java
entityErrorList = response.getItems().length > 0 ? response.getItems()
.stream()
.filter(BulkItemResponse::isFailed)
.map(
    item -> {
        int idx = Arrays.asList(response.getItems()).indexOf(item);
        EntityError().withMessage(itemResponse.getFailureMessage()).withEntity(request.get(idx).toString)
   }).toList() : entityErrorList
```

Also do you if `List<EntityError> entityErrorList` will be mutable and how it will behave with the multithreading?